### PR TITLE
Make animal health-related culling probability user-input

### LINKED ISSUE #1459: Make the breakdown of reasons for non-repro/production related culls a user input instead of hard coded
**Is your feature request related to a problem? Please describe.**
RuFaS used to have 2 different disease related culling inputs: 1) the ability to change the likelihood a cow was culled (removed from the herd) on a given day of lactation and 2) the % of total non-reproduction/milk production related culls associated with a given reason (ie. 16% feet and legs, 24% mastitis, etc.). The information requested in the first culling input is an array of probabilities adding up to 100%, with each probability corresponding to a specific days in milk. The second culling input was just a single number. Through a variety of refactors, the 1st input is the only one that still exists as changeable by the user and the 2nd one is hard coded. However, if we consider what data is accessible to users, the easier to obtain data is the single % (item 2), not the array of probabilities (item 1). 

**Describe the solution you'd like**
I would like to see user inputs associated with the total % probability that a cow is culled for a specific reason. From the paper we used they are as follows: 
Feet and legs 16.33
Injury 28.83
Mastitis 24.39
Disease 13.91
Udder  6.45
Unknown  10.09
These can exist as the user visualized defaults, but I would like to see them be editable by the user. 

**Describe alternatives you've considered**
Truthfully, I haven't considered very many alternatives to this, because I think it should be relatively easy to lift these from being hard coded to being user inputs but I'm open to suggestions.

### PR DESCRIPTION
**Pull request by allisterakun**
Make animal health-related culling probability a user-input
### Context

Issue(s) closed by this pull request: closes #1459 

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.

### What

- Updated the metadata property and the animal input JSON files to add the probability of each health-related culling reason as animal input.
    - Restructured the culling section:
      ```JSON
      "culling": {
		"cull_day_count": [0, 5, 15, 45, 90, 135, 180, 225, 270, 330, 380, 430, 480, 530],
		"feet_leg_cull": {
			"probability": 0.1633,
			"cull_day_prob": [0, 0.03, 0.08, 0.16, 0.25, 0.36, 0.48, 0.59, 0.69, 0.78, 0.85, 0.90, 0.95, 1]
		},
		"injury_cull": {
			"probability": 0.2883,
			"cull_day_prob": [0, 0.08, 0.18, 0.28, 0.38, 0.47, 0.56, 0.64, 0.71, 0.78, 0.85, 0.90, 0.95, 1]
		},
		"mastitis_cull": {
			"probability": 0.2439,
			"cull_day_prob": [0, 0.06, 0.12, 0.19, 0.30, 0.43, 0.56, 0.68, 0.78, 0.85, 0.90, 0.94, 0.97, 1]
		},
		"disease_cull": {
			"probability": 0.1391,
			"cull_day_prob": [0, 0.04, 0.12, 0.24, 0.34, 0.42, 0.50, 0.57, 0.64, 0.72, 0.81, 0.89, 0.95, 1]
		},
		"udder_cull": {
			"probability": 0.0645,
			"cull_day_prob": [0, 0.12, 0.24, 0.33, 0.41, 0.48, 0.55, 0.62, 0.68, 0.76, 0.82, 0.89, 0.95, 1]
		},
		"unknown_cull": {
			"probability": 0.1009,
			"cull_day_prob": [0, 0.05, 0.11, 0.18, 0.27, 0.37, 0.45, 0.54, 0.62, 0.70, 0.77, 0.84, 0.92, 1]
		},
		"parity_death_prob": [0.039,0.056,0.085,0.117],
		"parity_cull_prob": [0.169, 0.233, 0.301, 0.408],
 		"death_cull_prob": [0, 0.18, 0.32, 0.42, 0.48, 0.54, 0.60, 0.65, 0.70, 0.77, 0.83, 0.89, 0.95, 1]
	}
      ```
- Updated `Cow.health_cull_update()` function logic to adapt to the change.
### Why

To make animal health-related culling probability a user-input.

### CODE REVIEW

**Review by ew3361zh — APPROVED** [review]
Looks good, Allister, seemed to work as described. I'll leave it up to Kaitlyn on the default probabilities being in the ranges we're hoping for.

File: input/data/animal/barnyard_animal.json [hunk 1]
```diff
@@ -103,17 +103,34 @@
...
-				"parity_cull_prob": [0.169, 0.233, 0.301, 0.408],
-				"mastitis_cull_prob": [0, 0.06, 0.12, 0.19, 0.30, 0.43, 0.56, 0.68, 0.78, 0.85, 0.90, 0.94, 0.97, 1],
-				"feet_leg_cull_prob": [0, 0.03, 0.08, 0.16, 0.25, 0.36, 0.48, 0.59, 0.69, 0.78, 0.85, 0.90, 0.95, 1],
-				"injury_cull_prob": [0, 0.08, 0.18, 0.28, 0.38, 0.47, 0.56, 0.64, 0.71, 0.78, 0.85, 0.90, 0.95, 1],
-				"disease_cull_prob": [0, 0.04, 0.12, 0.24, 0.34, 0.42, 0.50, 0.57, 0.64, 0.72, 0.81, 0.89, 0.95, 1],
-				"udder_cull_prob": [0, 0.12, 0.24, 0.33, 0.41, 0.48, 0.55, 0.62, 0.68, 0.76, 0.82, 0.89, 0.95, 1],
-				"unknown_cull_prob": [0, 0.05, 0.11, 0.18, 0.27, 0.37, 0.45,
-					0.54, 0.62, 0.70, 0.77, 0.84, 0.92, 1],
-				"cull_day_count": [0, 5, 15, 45, 90, 135, 180, 225, 270, 330, 380, 430, 480, 530]
+				"death_cull_prob": [0, 0.18, 0.32, 0.42, 0.48, 0.54, 0.60, 0.65, 0.70, 0.77, 0.83, 0.89, 0.95, 1],
```
Comment (krb78) [inline input/data/animal/barnyard_animal.json]: can I just request that we change this one's name to "death_day_prob" (as long as you think that sounds ok), I want to just signal to users that this is the day array similar to the other culling reasons.

File: input/metadata/properties/default.json [hunk 2]
```diff
@@ -602,76 +602,144 @@
...
+              }
             }
           },
-          "udder_cull_prob": {
-            "type": "array",
-            "description": "Culling probability for udder reasons",
-            "properties": {
+          "disease_cull": {
+            "type": "object",
+            "description": "Cull probabilities due to disease-related health issues.",
```
Comment (krb78) [inline input/metadata/properties/default.json:672]: can you just add "other" in the description? So it reads as follows "cull probabilities due to other disease-related health issues"

File: input/metadata/properties/default.json [hunk 3]
```diff
@@ -602,76 +602,144 @@
...
+            "probability": {
               "type": "number",
+              "description": "The probability of culling due to disease-related health issues.",
               "minimum": 0,
               "maximum": 1
+            },
+            "cull_day_prob": {
+              "type": "array",
+              "description": "Culling probability on the cull day for disease-related health issues.",
```
Comment (krb78) [inline input/metadata/properties/default.json:681]: add "other" here as well :)

**Review by krb78 — APPROVED** [review]
looks great! thanks, Allister :) I just had a few minor wording edits 🐮

**Pull request merged**